WriteHDLUtils.py

```diff
@@ -265,6 +268,13 @@ def getListsOfGroupedMemories(aProcModule):
     memList = list(aProcModule.upstreams + aProcModule.downstreams)
     portList = list(aProcModule.input_port_names + aProcModule.output_port_names)
 
+    # Sort the lists using portList, first by the phi region number (e.g. 2 in "vmstuboutPHIA2"), then alphabetically
+    zipped_list = zip(memList, portList)
+    zipped_list.sort(key=lambda (m, p): 0 if ('PHI' not in p or not p[-1].isdigit()) else int("".join([i for i in p if i.isdigit()]))) # sort by number
```

Comment:
I don't understand what the point of the complicated sort using isdigit() is. If I delete this line, the results are unchanged.

Reply:
Without that line, it will think that 10 comes before 9, so I'm trying to extract the number after "PHIX" and turn it into an int before sorting. But you're right that it did nothing because I forgot to add stuff on the next line. I've fixed the bug now, but it did not get any less complicated :-)